### PR TITLE
Updated to not allow processing of worker commands embedded in git co…

### PR DESCRIPTION
**Pull request by jtpetty**
…mment

https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1645065

### CODE REVIEW

**Comment by jtpetty** [comment]
The updates to dotnet* files were because the tests were failing since they did not match the Internet versions

File: src/Agent.Sdk/ProcessInvoker.cs [hunk 1]
```diff
@@ -61,9 +64,10 @@ public void Reset()
         public event EventHandler<ProcessDataReceivedEventArgs> OutputDataReceived;
         public event EventHandler<ProcessDataReceivedEventArgs> ErrorDataReceived;
 
-        public ProcessInvoker(ITraceWriter trace)
+        public ProcessInvoker(ITraceWriter trace, bool disallowWorkerCommands = false)
```
Comment (stephenmichaelf) [inline src/Agent.Sdk/ProcessInvoker.cs]: Nit: disallow -> disable?

File: src/Agent.Plugins/GitCliManager.cs [hunk 2]
```diff
@@ -565,7 +565,7 @@ private async Task<int> ExecuteGitCommandAsync(AgentTaskPluginExecutionContext c
             string arg = StringUtil.Format($"{command} {options}").Trim();
             context.Command($"git {arg}");
 
-            var processInvoker = new ProcessInvoker(context);
+            var processInvoker = new ProcessInvoker(context, true);
```
Comment (stephenmichaelf) [inline src/Agent.Plugins/GitCliManager.cs]: Nit: can we make this a named parameter here to make it easier to see what's going on:

`var processInvoker = new ProcessInvoker(context, disallowWorkerCommands: true);`

Same for occurrences of it below.

File: src/Agent.Sdk/ProcessInvoker.cs [hunk 3]
```diff
@@ -20,6 +21,8 @@ namespace Microsoft.VisualStudio.Services.Agent.Util
     // The reason is we find a huge perf issue about process STDOUT/STDERR with those events.
     public sealed partial class ProcessInvoker : IDisposable
     {
+        public bool DisallowWorkerCommands {get; set; }
```
Comment (stephenmichaelf) [inline src/Agent.Sdk/ProcessInvoker.cs]: Nit: public after private to group with others.

File: src/Agent.Sdk/ProcessInvoker.cs [hunk 4]
```diff
@@ -485,6 +489,10 @@ private void StartReadStream(StreamReader reader, ConcurrentQueue<string> dataBu
                     string line = reader.ReadLine();
                     if (line != null)
                     {
+                        if (DisallowWorkerCommands)
+                        {
+                            line = Regex.Replace(line, "##vso", "##DISALLOWED-vso", RegexOptions.IgnoreCase);
```
Comment (stephenmichaelf) [inline src/Agent.Sdk/ProcessInvoker.cs]: Do we want to print like this or print an info message with the contents describing what happened and the original message? Not sure what is the best UX and what will help customers the most.

File: src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs [hunk 5]
```diff
@@ -113,10 +113,13 @@ public sealed class ProcessInvokerWrapper : AgentService, IProcessInvoker
     {
         private ProcessInvoker _invoker;
 
+        public bool DisallowWorkerCommands {get; set; }
+
```
Comment (stephenmichaelf) [inline src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs]: Nit: extra line.

File: src/Agent.Sdk/ProcessInvoker.cs (hunk 4 above)
Comment (vtbassmatt) [reply src/Agent.Sdk/ProcessInvoker.cs]: Yeah, this doesn't feel quite right. Maybe replace the `##vso` with `**vso` and append a `##warning` line explaining what happened?

Comment (jtpetty) [reply src/Agent.Sdk/ProcessInvoker.cs]: I changed "##vso" to "**vso" but am unable to inject a warning message with my current understanding of the code base

File: src/agentversion [hunk 6]
```diff
@@ -1 +1 @@
-2.161.0
+2.161.1
```
Comment (damccorm) [inline src/agentversion]: Not super important, but normally don't we keep this at `2.161.0` in master?

**Comment by vtbassmatt** [comment]
Tommy and I talked about it briefly. There's not really a good way to inject a message. Since there's not much of a use case for this to work anyhow, I think it's alright.

**Pull request merged**